taichi/rhi/vulkan/vulkan_device.cpp

```diff
@@ -1923,6 +1902,45 @@ Stream *VulkanDevice::get_compute_stream() {
   return iter->second.get();
 }
 
+void VulkanCommandList::begin_profiler_scope(const std::string &kernel_name) {
+  auto pool = vkapi::create_query_pool(ti_device_->vk_device());
+  vkCmdResetQueryPool(buffer_->buffer, pool->query_pool, 0, 2);
+  vkCmdWriteTimestamp(buffer_->buffer, VK_PIPELINE_STAGE_ALL_COMMANDS_BIT,
+                      pool->query_pool, 0);
+  ti_device_->profiler_add_sampler(kernel_name, pool);
```

Comment:
nit: also use this function to add debug markers for tools like renderdoc or NSight.

https://www.saschawillems.de/blog/2016/05/28/tutorial-on-using-vulkans-vk_ext_debug_marker-with-renderdoc/

Reply:
Thanks, we can do this in a follow-up PR